cloud_foundry_api/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,71 @@
+name: Cloud Foundry API
+files:
+- name: cloud_foundry_api.yaml
+  options:
+  - template: init_config
+    options:
+    - template: init_config/default
+  - template: instances
+    options:
+    - template: instances/default
+    - name: api_url
+      description: URL of the Cloud Controller API.
+      required: true
+      value:
+        type: string
+    - name: uaa_url
+      description: URL of the UAA server.
+      required: true
+      value:
+        type: string
+    - name: tls_verify
+      value:
+        example: true
+        type: boolean
+      description: Instructs the check to validate the TLS certificate of services.
+    - name: client_id
+      description: |
+        `client_id` used to authenticate with UAA and obtain an oauth token
+        for communication with the Cloud Controller API.
+      required: true
+      value:
+        type: string
+    - name: client_secret
+      description: |
+        `client_secret` used to authenticate with UAA and obtain an oauth token
+        for communication with the Cloud Controller API.
+      required: true
+      value:
+        type: string
+      secret: true
+    - name: api_version
+      description: |
+        Version of the Cloud Controller API to target. Choose between `v2` and `v3`.
+        Depending on your Cloud Controller version, not all resources are available in `v3`.
+        Switch to `v2` if that is the case.
+      value:
+        type: string
+        pattern: v2|v3
+        default: v3
+        example: v3
+    - name: event_filter
+      description: |
+        A list of the types of audit events to fetch.
+        See https://docs.cloudfoundry.org/running/managing-cf/audit-events.html#types for detailed information.
+      value:
+        type: array
+        items:
+          type: string
+        default: []
+        example:
+        - audit.app.restage
+        - audit.app.update
+        - audit.app.create
+    - name: results_per_page
```

Comment:
Could we remove this config and always fetch for the max?

Reply:
Not sure we want the max actually. I'm actually thinking the 2000 default is already too much.
Since we'll query every 15 second, we actually need a limited number of results per page since there shouldn't be too many results to go through. I'm thinking of reducing the default to 100.